test/framework/streaming/general.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var assert = require('../utils/assert.js');
+var Server = require('../utils/server.js');
+var preq = require('preq');
+
+// mocha defines to avoid JSHint breakage
+/* global describe, it, before, beforeEach, after, afterEach */
+
+describe('Handler Template', function() {
+    var server = new Server('test/framework/streaming/test_config.yaml');
+    before(function() { return server.start(); });
+
+    it('Basic streaming', function () {
+        return preq.get({ uri: server.hostPort + '/test/hello' })
+        .then(function (res) {
+            assert.deepEqual(res.status, 200);
+            assert.deepEqual(res.headers['content-type'], 'text/html');
+            assert.deepEqual(res.body, 'hello');
+        });
+    });
+
+    it('Buffer streaming', function () {
+        return preq.get({ uri: server.hostPort + '/test/buffer' })
+        .then(function (res) {
+            assert.deepEqual(res.status, 200);
+            assert.deepEqual(res.headers['content-type'], 'text/html');
+            assert.deepEqual(res.body, 'hello');
+        });
+    });
+
+    it('Buffer streaming, no compression', function () {
```

Comment:
This test is identical to the one above (except the `gzip: true` bit), so how can you tell / test that gzip wasn't used?